explainability-core/src/main/java/org/kie/trustyai/explainability/model/domain/FeatureDomain.java

```diff
@@ -51,4 +51,11 @@
      */
     Set<T> getCategories();
 
-}
\ No newline at end of file
+    /**
+     * Get a string representation of the domain
+     *
+     * @return the string representation
+     */
+    String toString();
```

Comment:
This method clashes with [`Object.toString()` method](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#toString--) and it is usually something to avoid. I would prefer to use a different one like `prettyPrint()`/`describe()` or also `asString()`